src/spider/worker/worker.cpp

```diff
@@ -123,15 +124,49 @@ auto get_environment_variable() -> absl::flat_hash_map<
     return environment_variables;
 }
 
+/**
+ * Checks if the task is cancelled. If the task state is set to cancelled, cancels the running task.
+ * @param conn The storage connection to use.
+ * @param metadata_store
+ * @param executor_handle Handle to the task id and executor.
+ */
+auto check_task_cancel(
+        std::shared_ptr<spider::core::StorageConnection> const& conn,
+        std::shared_ptr<spider::core::MetadataStorage> const& metadata_store,
+        spider::worker::ExecutorHandle& executor_handle
+) -> void {
+    std::optional<boost::uuids::uuid> const optional_task_id = executor_handle.get_task_id();
+    if (!optional_task_id.has_value()) {
+        return;
+    }
+    boost::uuids::uuid const task_id = optional_task_id.value();
+
+    spider::core::TaskState task_state = spider::core::TaskState::Running;
+    spider::core::StorageErr err = metadata_store->get_task_state(*conn, task_id, &task_state);
+    if (false == err.success()) {
+        spdlog::error("Failed to get task state: {}", err.description);
+        return;
+    }
+
+    if (spider::core::TaskState::Canceled != task_state) {
+        return;
+    }
+
+    // Cancel the task.
```

Comment:
As mentioned many times before, please avoid comments like this. I've linked the guidelines in other posts, but the bottom line is that if code is so confuing it needs a comment we should first try to refactor or rename it. If a single line of code is that confusing then it is almost certainly named incorrectly.